src/test/java/org/prebid/server/bidder/ogury/OguryBidderTest.java

```diff
@@ -0,0 +1,335 @@
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class OguryBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "https://test.endpoint.com";
+
+    @Mock
+    private CurrencyConversionService currencyConversionService;
+
+    private OguryBidder target;
+
+    @BeforeEach
+    public void setUp() {
+        target = new OguryBidder(ENDPOINT_URL, currencyConversionService, jacksonMapper);
+    }
+
+    @Test
+    public void shouldFailOnInvalidEndpointUrl() {
+        Assertions.assertThatIllegalArgumentException()
+                .isThrownBy(() -> new OguryBidder("invalid_url", currencyConversionService, jacksonMapper));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldEncodePassedBidRequest() {
+        // given
+        when(currencyConversionService.convertCurrency(any(), any(), any(), any())).thenReturn(BigDecimal.ONE);
+        final BidRequest bidRequest = givenBidRequest();
+        final BidRequest modifiedBidRequest = givenModifiedBidRequest();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        final MultiMap expectedHeaders = HttpUtil.headers()
+                .add(HttpUtil.USER_AGENT_HEADER, "ua")
+                .add(HttpUtil.ACCEPT_LANGUAGE_HEADER, "en-US")
+                .add(HttpUtil.X_FORWARDED_FOR_HEADER, "0.0.0.0")
+                .add(HttpUtil.X_FORWARDED_FOR_HEADER, "ip6");
+        final Result<List<HttpRequest<BidRequest>>> expectedResult = Result.withValue(HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(ENDPOINT_URL)
+                .headers(expectedHeaders)
+                .impIds(Set.of("imp_id"))
+                .body(jacksonMapper.encodeToBytes(modifiedBidRequest))
+                .payload(modifiedBidRequest)
+                .build());
+        assertThat(result.getValue()).usingRecursiveComparison().isEqualTo(expectedResult.getValue());
+        assertThat(result.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenRequestDoesNotHaveOguryKeys() {
+        // given
+        final ObjectNode bidder = mapper.createObjectNode();
+        bidder.putIfAbsent("bidder", mapper.createObjectNode());
+        final BidRequest bidrequest = givenBidRequest(bidRequest ->
+                bidRequest.imp(List.of(Imp.builder().ext(bidder).build())));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidrequest);
+
+        // then
+        assertThat(result.getErrors()).isNotEmpty()
+                .contains(BidderError.badInput(
+                        "Invalid request. assetKey/adUnitId or request.site.publisher.id required"));
+    }
```

Comment:
Add more tests that cover scenarios:
- impWithOguryParams + site.publisher.id + impWithoutOguryParams
- invalid impWithOguryParams  + site.publisher.id + impWithoutOguryParams
- impWithoutOguryParams
- imp.tagId copied from imp.id
- price tests (for each edge case)
- test that checks imp.ext fields aren't lost, and new fields are added/replaced by impExtBidderHoist fields